src/prototype_room_my.js

```diff
@@ -346,6 +346,35 @@ Room.prototype.executeRoom = function() {
     }
   }
 
+  let carryHelpInterval = config.nextRoom.ticksUntilHelpCheck;
+  if (Game.time % carryHelpInterval) {
+    this.memory.Energy += this.energyAvailable;
+  } else if (!this.memory.Energy) {
+    this.memory.Energy = 0;
+  } else {
+    let storage = this.storage;
+    if (this.memory.Energy < 300 * carryHelpInterval) {
+      Memory.needEnergy.push(this.name);
+      console.log('!!!', this.name, ' need energy !!!');
+    } else if (this.memory.Energy > 1000 * carryHelpInterval && storage.length) {
+      this.memory.queue.push({
```

Comment:
Would use `this.checkRoleToSpawn('carry', 1, storage[0].id, this.name)` instead.